### PR TITLE
CONTRIBUTING: Add code formatting instructions

### PR DESCRIPTION
**Pull request by Javyre**
Added some instructions on how to format the code before committing.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,6 +34,16 @@ If you want to work on a feature or a bug fix which does not have an open issue,
...
 
+### Code Formatting
+
+Please use [`uncrustify`](https://github.com/uncrustify/uncrustify) to automatically format the code before committing:
+
+```sh
+$ git ls-files | grep "hpp$\|cpp$" | xargs uncrustify -c uncrustify.ini --no-backup
+```
+
+You can setup a [githook](https://githooks.com/) to run this automatically before committing.
```
Comment (soreau) [inline CONTRIBUTING.md:45]: I would use https://git-scm.com/docs/githooks or make it even more reliable and use no link at all.

Comment (Javyre) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -34,6 +34,16 @@ If you want to work on a feature or a bug fix which does not have an open issue,
 
 In any case, feel free to ask questions if you do not understand a part of the code, or if you are unsure how a particular feature should be implemented.
 
+### Code Formatting
+
+Please use [`uncrustify`](https://github.com/uncrustify/uncrustify) to automatically format the code before committing:
```
Comment (soreau) [inline CONTRIBUTING.md:39]: We might want to say somewhere that the version of uncrustify should be >= 0.7.1.

Comment (Javyre) [reply CONTRIBUTING.md]: alright I added it in parentheses

**Review by soreau — APPROVED** [review]
Thanks for working on this!

**Review by ammen99 — APPROVED** [review]
Thanks!

**Pull request merged**